### PR TITLE
GPU type selectors

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Adds support for specifying extra metadata associated with task resources (e.g. accelerator type - gpu device, partition size etc.)

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: tests/flytekit/unit/core/test_node_creation.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
 from collections import OrderedDict
 from dataclasses import dataclass
 
-import pytest
-
 import flytekit.configuration
+import pytest
 from flytekit import Resources, map_task
 from flytekit.configuration import Image, ImageConfig
+from flytekit.core.accelerators import NvidiaTeslaA100, NvidiaTeslaT4
```
Comment (kumare3) [inline tests/flytekit/unit/core/test_node_creation.py:10]: can we move it to extras?
